subsys/mgmt/osdp/Kconfig

```diff
@@ -56,6 +56,36 @@ config OSDP_PACKET_TRACE
 	  Prints bytes sent/received over OSDP to console for debugging.
 	  LOG_HEXDUMP_DBG() is used to achieve this and can be very verbose.
 
+config OSDP_SC_ENABLED
+	bool "Enable OSDP Secure Channel"
+	default y
+	select CRYPTO
+	select CRYPTO_MBEDTLS_SHIM
+	select MBEDTLS
+	select MBEDTLS_CIPHER_CCM_ENABLED
+	select TEST_RANDOM_GENERATOR
```

Comment:
You should not select the test random generator, it is only meaningful in testing and should be only selected by sample prj.conf if needed.

Reply:
When user selects Secure Channel feature in menuconfig, there must be a random function defined and available for build to succeed. I was having it in my `prj.conf` file during development but moved it here for better usability. What is the right way to ask for randomness in Zephyr?

Reply:
@cbsiddharth, let the platform definition select the RANDOM_GENERATOR. If there isn't one then there isn't a true RNG support for the platform.

Reply:
@dleach02, Thank you. For now, I'll put `CONFIG_TEST_RANDOM_GENERATOR=y` in the samples's prj.conf.

Reply:
what platform are you testing on that causes you to add this?

Reply:
I use a STM32F1x (board/stm32_min_dev) for development and it does not have a TRNG.

Reply:
Thinking about it, shouldn't zephyr have a way to auto determine the most secure way to generate randomness based on platform capabilities?

This seems like a reasonable ask for me; but I'm just beginning to familiarizing myself with Zephyr.

Reply:
We have these types of discussions in the security working group. There are two types of random numbers being needed by an application; cryptographic and non-cryptographic. With non-cryptographic using the TEST_RANDOM_GENERATOR or a timer based one for seed into XOROSHIRO algorithm is fine. If the application/subsystem needs a cryptographically secure RNG value then it should use the csrand functions which will propagate down to only building with valid entropy sources (TRNG). If the STM32F1x doesn't have HW entropy then technically speaking it shouldn't be allowed to use cryptographic functions. 

Some of this groundwork has started with the addition of csrand based functions. 

https://docs.zephyrproject.org/latest/reference/random/index.html

Reply:
> We have these types of discussions in the security working group

Is this a slack channel or mailing list? I'd love to listen-in.

> If the STM32F1x doesn't have HW entropy then technically speaking it shouldn't be allowed to use cryptographic functions.

I agree. But shouldn't there be a fallback mechanism? I'd love to use the csrand methods, but still be able to do a spot check on a STM32F1x as I have hooked up 2 of them together to test both my CP and PD modes before publishing code upstream.

Basically, I feel a wrapper around `sys_rand_get` and `sys_csrand_get` based on platform capabilities (choosing the most secure option available) so application code can remain unchanged.

Reply:
The TEST_RANDOM_GENERATOR is for test cases to use and not a fallback for actual end products. There is a security slack channel to engage in as well.

> Basically, I feel a wrapper around sys_rand_get and sys_csrand_get based on platform capabilities (choosing the most secure option available) is needed so application code can remain unchanged.

Definitely not an option. You do not want to present a false sense of security on platforms that can't support it. The objective of the rand split was to start having improving security profile on applications and platforms built with Zephyr. If the platform doesn't have a cryptographic entropy source then it should not host any protocols or applications that claim to have security.

Now at some point someone might add support for a software based entropy generation but there has been no work on that so far. Most MCU devices have either a true hardware source for random values or a HW source to retrieve entropy values to seed a CSPRNG function.

Reply:
Okay.